### ISSUE TITLE
apr 28

### ISSUE THREAD

**Issue by hillarysanders** [hillarysanders/notes#22]
This week I ...
- [x] added a bunch of modules and unit tests to Rich's utils repo
- [x] suped up my tensorboard callback in keras - moar-plots!
going slow due
- [x] looked into encoding used in URLs, still not totally sure of things there.
- [x] testing out convnet on sophos data, comparing FPRs - can't compare detection rates yet because we only have benign data.
- [x]  started looking into how to set up an API on AWS so people can beta test our models via API pings (will need to be secure)

TODO:
minis:
- [ ] 1. start test_fitted_model.py on most recent convnet run (sophos data). Then do the same for validation data from the original dataset (just copy the 40m dataset, delete all training obs, and run the python code on the validation obs). Then (later) compare FPRs by various score cutoffs (and their related detection rates in VT) between sophos vs VT data. [HOLDING OFF FOR NOW UNTIL MODEL IS DONE TRAINING ON THE DEEP-LEARN INSTANCE, SINCE RESTARTING HASN'T BEEN WORKING WELL]
- [x] 2. data fetch seems to have crashed - add code to restart automagically and restart
bigger things:  
- [ ] make slides explaining a feed forward pass of a neural network :)!!! Nice!
- [ ] compare FPR rate in VT (training dataset, validation set) data to FPR rate in Sophos benign data on final fitted convnet.
- [ ] learn about Lambdas on AWS,  spin one up to create a tiny working API
   - [ ] create instructions or scripts to create one of these
    - [ ] fix permissions
    - [ ] add in a working model (e.g. convnet), test out
- [ ] encoding of URL characters

**Comment by hillarysanders** [comment]
Increase the difference between the weights on the benign and malicious files. 
sigmoid or softmax was saturating. So the values are so large that when you apply 32 bit floats the 99.9% bad turns to 1, and that can never really be changed. Applied it to the HTML model. Better! Bumps up AUC! Nice.
